airflow/www/views.py

```diff
@@ -1293,24 +1303,30 @@ def set_duration(tid):
     def graph(self, session=None):
         dag_id = request.args.get('dag_id')
         blur = conf.getboolean('webserver', 'demo_mode')
-        dag = dagbag.get_dag(dag_id)
-        if dag_id not in dagbag.dags:
-            flash('DAG "{0}" seems to be missing.'.format(dag_id), "error")
-            return redirect('/')
-
         root = request.args.get('root')
         if root:
-            dag = dag.sub_dag(
-                task_regex=root,
-                include_upstream=True,
-                include_downstream=False)
+            show_dag_id = root
+        else:
+            show_dag_id = dag_id
+        dag = session.query(models.DagModel).filter(models.DagModel.dag_id == show_dag_id).first()
+        if not dag:
+            flash('DAG "{0}" seems to be missing.'.format(show_dag_id), "error")
+            return redirect('/admin/')
 
-        arrange = request.args.get('arrange', dag.orientation)
+        arrange = request.args.get('arrange', configuration.conf.get('webserver', 'dag_orientation'))
+        dt_nr_dr_data = get_date_time_num_runs_dag_runs_form_data(request, session, dag)
+        dt_nr_dr_data['arrange'] = arrange
+        dttm = dt_nr_dr_data['dttm']
 
-        nodes = []
-        edges = []
-        for task in dag.tasks:
-            nodes.append({
+        dag_run = session.query(models.DagRun) \
+            .filter(models.DagRun.dag_id == show_dag_id) \
+            .filter(models.DagRun.execution_date == dttm).one()
```

Comment:
This would be nice as a method on `dag` (which if we define a FK between Dag and DagRun then would be something like `dag.dag_runs.filter(models.DagRun.execution_date == dttm).one()`)

Reply:
I do agree but since FK's are almost not used in airflow. This should be done over the complete database I think